src/resources/ReservationResource.ts

```diff
@@ -0,0 +1,40 @@
+import JSONAPIModel from "../util/json-api/JSONAPIModel";
+
+export interface ReservationAttributes {
+    'is-partial-day': boolean;
```

Comment:
I thought about making this fluent, but wasn't sure what the best way to map response fields to variable in typescript. I Kotlin, I would name a field `isPartialDay` and then annotate it with a `@JsonProperty('is-partial-day')`. This seems possible in typescript, but I would need to add a library called `class-transformed` with the `@Expose` annotation. I don't yet feel proficient enough in typescript to decide that this is the best approach, so I have avoided doing it for now.